src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/layout/fetch/fetchFormLayoutSagas.ts

```diff
@@ -1,30 +1,46 @@
 import { SagaIterator } from 'redux-saga';
 import { call, all, take, select } from 'redux-saga/effects';
-import { IAltinnWindow } from 'altinn-shared/types';
-import { getLayoutSettingsUrl } from 'src/utils/urlHelper';
+import { IAltinnWindow, IInstance } from 'altinn-shared/types';
+import { IApplicationMetadata } from 'src/shared/resources/applicationMetadata';
+import { getLayoutSettingsUrl, getLayoutSetsUrl } from 'src/utils/urlHelper';
 import { get } from '../../../../utils/networking';
 import Actions from '../formLayoutActions';
 import { IFetchFormLayout } from './fetchFormLayoutActions';
 import * as ActionTypes from '../formLayoutActionTypes';
 import * as FormDataActionTypes from '../../data/formDataActionTypes';
 import QueueActions from '../../../../shared/resources/queue/queueActions';
 import { getRepeatingGroups } from '../../../../utils/formLayout';
-import { ILayoutSettings, IRuntimeState } from '../../../../types';
+import { ILayoutSettings, IRuntimeState, ILayoutSets } from '../../../../types';
 import { IFormDataState } from '../../data/formDataReducer';
 import { ILayouts } from '../index';
+import { getLayouytsetForDataElement } from '../../../../utils/layout';
+import { getDataTaskDataTypeId } from '../../../../utils/appMetadata';
 
 const formDataSelector = (state: IRuntimeState) => state.formData;
+const layoutSetsSelector = (state: IRuntimeState) => state.formLayout.layoutset;
+const instanceSelector = (state: IRuntimeState) => state.instanceData.instance;
+const applicationMetadataSelector = (state: IRuntimeState) => state.applicationMetadata.applicationMetadata;
 
 function* fetchFormLayoutSaga({ url }: IFetchFormLayout): SagaIterator {
   try {
-    const layoutResponse: any = yield call(get, url);
+    const { org, app } = window as Window as IAltinnWindow;
+    const layoutSets: ILayoutSets = yield select(layoutSetsSelector);
+    const instance: IInstance = yield select(instanceSelector);
+    const formDataState: IFormDataState = yield select(formDataSelector);
+    const aplicationMetadataState: IApplicationMetadata = yield select(applicationMetadataSelector);
+    const dataType: string = getDataTaskDataTypeId(instance.process.currentTask.elementId,
+      aplicationMetadataState.dataTypes);
+    let apiUrl: string = url;
+    if (layoutSets != null) {
+      const layoutSetId: string = getLayouytsetForDataElement(instance, dataType, layoutSets);
+      apiUrl = `${window.location.origin}/${org}/${app}/api/layouts/${layoutSetId}`;
```

Comment:
Perhaps move this url into the urlHelper as `getLayoutsUrl`? With an optional `layoutSets` parameter like you have done for `getLayoutSettingsUrl`?
Then you could also remove the `url` parameter from the action call, which is something we have been trying to do for the other actions :)

Reply:
done